lapack/src/KokkosLapack_svd.hpp

```diff
@@ -0,0 +1,191 @@
+  static_assert(Kokkos::is_view<UMatrix>::value,
+                "KokkosLapack::svd: U must be a Kokkos::View.");
+  static_assert(Kokkos::is_view<VMatrix>::value,
+                "KokkosLapack::svd: Vt must be a Kokkos::View.");
+  static_assert(AMatrix::rank() == 2, "KokkosLapack::svd: A must have rank 2.");
+  static_assert(SVector::rank() == 1, "KokkosLapack::svd: S must have rank 1.");
+  static_assert(UMatrix::rank() == 2, "KokkosLapack::svd: U must have rank 2.");
+  static_assert(VMatrix::rank() == 2,
+                "KokkosLapack::svd: Vt must have rank 2.");
+
+  int64_t m     = A.extent(0);
+  int64_t n     = A.extent(1);
+  int64_t rankA = Kokkos::min(m, n);
+
+  // Check validity of output views sizes
+  bool is_extent_invalid = false;
+  std::ostringstream os;
+  if (S.extent_int(0) != rankA) {
+    is_extent_invalid = true;
+    os << "KokkosLapack::svd: S has extent " << S.extent(0) << ", instead of "
+       << rankA << ".\n";
+  }
+  if (U.extent_int(0) != m || U.extent_int(1) != m) {
+    is_extent_invalid = true;
+    os << "KokkosLapack::svd: U has extents (" << U.extent(0) << ", "
+       << U.extent(1) << ") instead of (" << m << ", " << m << ").\n";
+  }
+  if (Vt.extent_int(0) != n || Vt.extent_int(1) != n) {
+    is_extent_invalid = true;
+    os << "KokkosLapack::svd: V has extents (" << Vt.extent(0) << ", "
+       << Vt.extent(1) << ") instead of (" << n << ", " << n << ").\n";
+  }
+  if (is_extent_invalid) {
+    KokkosKernels::Impl::throw_runtime_exception(os.str());
+  }
+
+  // Check the jobu and jobvt control flags
+  // The only valid options there are 'A', 'S', 'O' and 'N'
+  const bool is_jobu_invalid =
+      !((jobu[0] == 'A') || (jobu[0] == 'a') || (jobu[0] == 'S') ||
+        (jobu[0] == 's') || (jobu[0] == 'O') || (jobu[0] == 'o') ||
+        (jobu[0] == 'N') || (jobu[0] == 'n'));
+
+  const bool is_jobvt_invalid =
+      !((jobvt[0] == 'A') || (jobvt[0] == 'a') || (jobvt[0] == 'S') ||
+        (jobvt[0] == 's') || (jobvt[0] == 'O') || (jobvt[0] == 'o') ||
+        (jobvt[0] == 'N') || (jobvt[0] == 'n'));
+
+  if (is_jobu_invalid && is_jobvt_invalid) {
+    std::ostringstream oss;
+    oss << "KokkosLapack::svd: both jobu and jobvt are invalid!\n"
+        << "Possible values are A, S, O or N, submitted values are " << jobu[0]
+        << " and " << jobvt[0] << "\n";
+    KokkosKernels::Impl::throw_runtime_exception(os.str());
+  }
+  if (is_jobu_invalid) {
+    std::ostringstream oss;
+    oss << "KokkosLapack::svd: jobu is invalid!\n"
+        << "Possible values are A, S, O or N, submitted value is " << jobu[0]
+        << "\n";
+    KokkosKernels::Impl::throw_runtime_exception(os.str());
+  }
+  if (is_jobvt_invalid) {
+    std::ostringstream oss;
+    oss << "KokkosLapack::svd: jobvt is invalid!\n"
+        << "Possible values are A, S, O or N, submitted value is " << jobvt[0]
+        << "\n";
+    KokkosKernels::Impl::throw_runtime_exception(os.str());
+  }
+
```

Comment:
Should we need to check sizes of ```U``` and ```Vt``` with respect to ```jobu``` and ```jobvt```, after the above steps?

Reply:
Done